### PR TITLE
Mock FrameTrackListModel in  SamplingWithFrameTrackInputWidgetTest

### PR DESCRIPTION
**Pull request by akopich**
Test: Unit
Bug: http://b/242839305

### CODE REVIEW

File: src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp [hunk 1]
```diff
@@ -52,6 +50,27 @@ class MockPairedData {
...
+
+static constexpr size_t kFrameTracksCount = 4;
+static constexpr std::array<FrameTrackId, kFrameTracksCount> kScopeIds = {
+    FrameTrackId(ScopeId(2)), FrameTrackId(ScopeId(1)), FrameTrackId(ScopeId(10)),
+    FrameTrackId(PresentEvent::kD3d9)};
+
+class MockFrameTrackListModel : public QAbstractListModel {
+ public:
+  MockFrameTrackListModel(const MockPairedData*, const absl::flat_hash_set<TID>*,
+                          const RelativeTimeNs*, QObject*) {}
```
Comment (beckerhe) [inline src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp:62]: The QObject parent should be passed on to the base class.

File: src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp [hunk 2]
```diff
@@ -52,6 +50,27 @@ class MockPairedData {
...
+
+  [[nodiscard]] int rowCount(const QModelIndex& /*parent*/) const override {
+    return kFrameTracksCount;
```
Comment (beckerhe) [inline src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp:65]: You still need to check whether `parent` is valid. If it's valid then you have to return 0, telling the view that a particular list entry has no children.

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp (hunk 2 above)
Comment (akopich) [reply src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp:65]: As discussed offline, this should be fixed in production implementations of models. The mock is fine as is.

**Review by danielfenner — APPROVED** [review]
lgtm

File: src/MizarModels/include/MizarModels/FrameTrackListModel.h [hunk 3]
```diff
@@ -131,6 +133,9 @@ class FrameTrackListModelTmpl : public QAbstractListModel {
   std::vector<FrameTrack> frame_tracks_;
 };
 
+using FrameTrackListModel =
```
[the remark this answers is not in the document]
Comment (akopich) [reply src/MizarModels/include/MizarModels/FrameTrackListModel.h:136]: As with most (all?) `*Tmpl` classes in Mizar, it has two instantiations. One is defined here and is supposed to be used in prod (I should have added a comment on that). The other one is used in in FrameTrackListModelTest.cpp -- there we instantiate with Mock classes.

**Pull request merged**